typha/pkg/discovery/discovery.go

```diff
@@ -214,40 +215,55 @@ func (d *Discoverer) discoverTyphaAddrs() ([]Typha, error) {
 
 	// If we get here, we need to look up the Typha service endpoints using the k8s API.
 	logrus.Info("(Re)discovering Typha endpoints using the Kubernetes API...")
-	epClient := d.k8sClient.CoreV1().Endpoints(d.k8sNamespace)
-	eps, err := epClient.Get(context.Background(), d.k8sServiceName, v1.GetOptions{})
+	epClient := d.k8sClient.DiscoveryV1().EndpointSlices(d.k8sNamespace)
+	epsList, err := epClient.List(context.Background(), v1.ListOptions{})
```

Comment:
If the service name is always present in a label on the EPSlice, I think it would be good to use a LabelSelector for the query here instead of querying for all slices and filtering them out afterwards.